### PR TITLE
3rd Update - Switch NetworkDisplay with PickerNetwork

### PR DESCRIPTION
**Pull request by MaxwellDG**
This is fixing a 3nd instance of https://github.com/MetaMask/metamask-extension/issues/20485.

As stated by @georgewrmarshall in https://github.com/MetaMask/metamask-extension/issues/20485 there are some expected, and accepted UI differences:

The colourful circle now includes a letter when it's not an image
The height is slightly different. They're now all static at 32px

Initial:
Network_3_Initial

Current:
Network_3_Current

### CODE REVIEW

File: ui/components/app/signature-request/signature-request-header/signature-request-header.component.js [hunk 1]
```diff
@@ -24,9 +46,33 @@ export default class SignatureRequestHeader extends PureComponent {
...
+            as="div"
+            src={currentNetwork?.rpcPrefs?.imageUrl}
+            label={
+              providerConfig?.type === NETWORK_TYPES.RPC
+                ? providerConfig?.nickname ?? t('privateNetwork')
+                : t(getNetworkLabelKey(providerConfig?.type))
+            }
+            backgroundColor={BackgroundColor.transparent}
+            borderColor={BorderColor.borderMuted}
+            iconProps={{ display: Display.None }}
```
Comment (brad-decker) [inline ui/components/app/signature-request/signature-request-header/signature-request-header.component.js:59]: @MaxwellDG @georgewrmarshall Do we have icons for our built in networks? I don't know if we want to introduce the letters here? Might cause user confusion.

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/components/app/signature-request/signature-request-header/signature-request-header.component.js (hunk 1 above)
Comment (MaxwellDG) [reply ui/components/app/signature-request/signature-request-header/signature-request-header.component.js:59]: I don't believe there's icons for the built-in ones. The previous components just used small coloured circles. And I couldn't see any Sepolia or Goerli related images in Storybook. Maybe you know of some elsewhere you'd like me to use?

Comment (georgewrmarshall) [reply ui/components/app/signature-request/signature-request-header/signature-request-header.component.js:59]: I believe the icons are retrieved here 
```
src={currentNetwork?.rpcPrefs?.imageUrl}
```

Sepolia and Goreli test networks use a combination of html elements and color to create the icon but this adds complexity to how we display networks. I'll create a ticket to add the images and migrate them over. This should reduce the complexity if all networks are displayed using images.

Comment (georgewrmarshall) [reply ui/components/app/signature-request/signature-request-header/signature-request-header.component.js:59]: Created an issue here https://github.com/MetaMask/metamask-extension/issues/21142

**Review by georgewrmarshall — COMMENTED** [review]
The code is looking good but I'm not sure this component is used anywhere so instead of replacing the `NetworkDisplay` component we may be able to remove this folder(`ui/components/app/signature-request-header`) completely. Apologies for not catching this earlier. The good news is by removing these component files we will get some small performance benefits in prod and dev because we are also removing redundant CSS, tests and storybook files. Let's get confirmation from @matthewwalsh0 on the confirmations team to see if we can remove them.

**Comment by georgewrmarshall** [comment]
Hey @MaxwellDG, Can you update the snapshots for this PR too?

**Pull request closed without merging**